api/dataset.go

```diff
@@ -680,6 +680,75 @@ func (api *DatasetAPI) getDimensionOptions(w http.ResponseWriter, r *http.Reques
 	log.Debug("get dimension options", log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID, "dimension": dimension})
 }
 
+func (api *DatasetAPI) getMetaData(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	datasetID := vars["id"]
+	edition := vars["edition"]
+	version := vars["version"]
+
+	// get dataset document
+	datasetDoc, err := api.dataStore.Backend.GetDataset(datasetID)
+	if err != nil {
+		log.Error(err, log.Data{"dataset_id": datasetID, "edition": edition, "version": version})
+		handleErrorType(versionDocType, err, w)
+		return
+	}
+
+	// Default state to published
+	state := models.PublishedState
+
+	// if request is authenticated then access resources of state other than published
+	if r.Header.Get(internalToken) == api.internalToken {
+		state = ""
+	}
+
+	if state == models.PublishedState {
+		// Check for current sub document
+		if datasetDoc.Current == nil || datasetDoc.Current.State != models.PublishedState {
+			log.ErrorC("found dataset but currently unpublished", errs.ErrDatasetNotFound, log.Data{"dataset_id": datasetID, "edition": edition, "version": version, "dataset": datasetDoc.Current})
+			http.Error(w, errs.ErrDatasetNotFound.Error(), http.StatusBadRequest)
+			return
+		}
+	}
+
+	if err = api.dataStore.Backend.CheckEditionExists(datasetID, edition, state); err != nil {
```

Comment:
If there was no check, the API wouldn't know if the request was a bad request or not found. 

If I removed this then it will not abide by the spec; if I change the spec for this endpoint then the spec for this endpoint is inconsistent with the other specs which inturn return bad request when the url does not make sense (can't have a version if there is no edition).

Reply:
👍  no change required